spec/vmfloaty/ssh_spec.rb

```diff
@@ -4,24 +4,30 @@
 require 'vmfloaty/ssh'
 
 class ServiceStub
-  def retrieve(_verbose, os_types, _use_token)
+  def retrieve(_verbose, os_types, _use_token, ondemand)
     if os_types.keys[0] == 'abs_host_string'
       return {
         os_types.keys[0] => { 'hostname' => ['abs-hostname.delivery.puppetlabs.net'] },
         'ok' => true
       }
+
+    elsif os_types.keys[0] == 'vmpooler_api_v2_host_string'
+      return {
+        os_types.keys[0] => { 'hostname' => ['vmpooler-v2-hostname.delivery.puppetlabs.net'] },
+        'ok' => true
+      }
     end
 
     {
-      os_types.keys[0] => { 'hostname' => 'vmpooler-hostname' },
+      os_types.keys[0] => { 'hostname' => 'vmpooler-v1-hostname.delivery.puppetlabs.net' },
```

Comment:
the hostname for V1 should not be an FQDN since the domain is passed a separate key. Not sure why the tests pass with this here?

Reply:
Oh that is odd, will look into that.

Reply:
Ah found it, seems that it was the missing `else` block for the last os_type. Now if the domain key is removed from the v1 hostname, then the test fails with the domain missing as expected.

However, there's a new code path that I'd like to add to testing as well and will push shortly.